### ISSUE TITLE
Updating just a stackName hangs after rollback_compete state

### ISSUE THREAD

**Issue by ddbhagat** [serverless/serverless#11295]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

- Deploy serverless changes
- Ensure the changes are deployed successfully
- Just update the stackName under serverless.yml. Say a minaor change from my-stack to my-new-stack
- Try to deploy this change

The changes never gets executed. It hungs after the step ROLLBACK_COMPLETE. And nothing happens after that. 

### Service configuration (serverless.yml) content

```yaml
NA
```

### Command name and used flags

npx sls deploy --stage [stage]

### Command output

```shell
The changes never gets executed. It hungs after the step ROLLBACK_COMPLETE. And nothing happens after that.
```

### Environment information

```shell
Framework Core: 1.67.3
Plugin: 3.6.6
SDK: 2.3.0
Components: 2.29.1
```

**Comment by pgrzesik** [comment]
Hello @ddbhagat, you're using a deprecation version of the Framework, yet you marked that you're using latest release. Why is that? Could you reproduce the problem on the latest release?

**Comment by ddbhagat** [comment]
@pgrzesik Here is the version that I am using, corrected above, I pasted incorrect version earlier

Framework Core: 3.16.0 (local)
Plugin: 6.2.2
SDK: 4.3.2

**Comment by pgrzesik** [comment]
Could you provide a small reproducible example? Also, what is the reason behind changing `stackName`? I would recommend first removing the original one and then deploying a new service with updated stack name, there might be some conflicts between resources in such case.

**Comment by ddbhagat** [comment]
Yeah that can be, delete and reconstruct. But do we have list of immutable properties? I thought there should be some guidelines of some of those attributes are not editable on existing stack. Thoughts ?

**Comment by ddbhagat** [comment]
i did try remove and deploy. But Now delete stack is failing as there is a custom domain associated with it, which is not created using serverless stack. Should that really be failed because of the resource which is not created by serverless at first place.

```
Error:
DELETE_FAILED: ApiGatewayRestApi (AWS::ApiGateway::RestApi)
Deleting RestApi [name] failed. Please remove all base path mappings related to the RestApi in your domains:
```

**Comment by pgrzesik** [comment]
Hey @ddbhagat - there is not such list and not necessarily such change will cause troubles, but I can expect that might be the case.

As for your other problem - if any change was done outside of serverless deployment, it would have to be reverted manually first. Serverless Framework cannot reliably deal with potential changes done on the side manually.
